File: frontend/src/types/skill.ts
export type ProficiencyLevel = "beginner" | "intermediate" | "advanced" | "expert";

export interface Skill {
  id: string;
  user_id: string;
  name: string;
  category: string | null;
  proficiency: ProficiencyLevel;
  description: string | null;
  created_at: string;
  updated_at: string;
}

export interface SkillFormData {
  name: string;
  category: string;
  proficiency: ProficiencyLevel;
  description: string;
}

export const PROFICIENCY_OPTIONS: { value: ProficiencyLevel; label: string }[] = [
  { value: "beginner", label: "Beginner" },
  { value: "intermediate", label: "Intermediate" },
  { value: "advanced", label: "Advanced" },
  { value: "expert", label: "Expert" },
];

export const EMPTY_SKILL_FORM: SkillFormData = {
  name: "",
  category: "",
  proficiency: "intermediate",
  description: "",
};

File: backend/app/schemas/skill.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field

from app.models.skill import ProficiencyLevel


class SkillBase(BaseModel):
    name: str = Field(..., min_length=1, max_length=100)
    category: str | None = Field(None, max_length=100)
    proficiency: ProficiencyLevel = ProficiencyLevel.INTERMEDIATE
    description: str | None = None


class SkillCreate(SkillBase):
    user_id: uuid.UUID


class SkillUpdate(BaseModel):
    name: str | None = Field(None, min_length=1, max_length=100)
    category: str | None = Field(None, max_length=100)
    proficiency: ProficiencyLevel | None = None
    description: str | None = None


class SkillResponse(SkillBase):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    user_id: uuid.UUID
    created_at: datetime
    updated_at: datetime
